3574/1/read_file.rb

```diff
@@ -0,0 +1,18 @@
+require 'yaml'
+
+class ReadFile
+  def call(from_file)
+    read_file(from_file)
+  end
+
+  private
+
+  def read_file(from_file)
+    if from_file[:gem_name]
+      gem_name = YAML.load_file(from_file[:file_name])['gems']
```

Comment:
seems that this line is common for both branches of `if` statement. please move it outside `if` statement

Reply:
and choose better name  for variable. Maybe gems_list/gem_names?